plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -145,6 +145,7 @@ public MachineRuntimeInfoImpl getRuntime() {
             try {
                 final ContainerInfo containerInfo = docker.inspectContainer(container);
                 machineRuntime = new MachineRuntimeInfoImpl(dockerMachineFactory.createMetadata(containerInfo,
+                                                                                                null,
```

Comment:
Feel free to add more details to your questions if that was not what you meant. I think you are wondering how `Server.url` is ever set if this parameter is always `null`. It should be set using property `machine.docker.local_node_host.external` but it looks as it's always null.

Property `machine.docker.local_node_host.external` is injected in `LocalDockerModule` objects that are binded to `DockerInstanceRuntimeInfo` in `LocalDockerModule`. 

Therefore method `DockerInstance.getRuntime()` is not used by the API to get the machine runtime. That explains why `Server.url` is correctly returned by the API.